diff -up jxrlib/image/sys/common.h.1~ jxrlib/image/sys/common.h
--- jxrlib/image/sys/common.h.1~	2021-03-05 05:26:15.765718571 +0100
+++ jxrlib/image/sys/common.h	2021-03-05 05:27:35.678388654 +0100
@@ -124,8 +124,8 @@ Void Adapt (CAdaptiveHuffman *pAdHuff, B
 Void AdaptFixed (CAdaptiveHuffman *pAdHuff);
 Void AdaptDiscriminant (CAdaptiveHuffman *pAdHuff);
 
-#ifndef _PREFAST_
-#pragma warning(disable:4068)
-#endif
+//#ifndef _PREFAST_
+//#pragma warning(disable:4068)
+//#endif
 
 #endif  // WMI_COMMON_H
diff -up jxrlib/image/sys/strcodec.c.1~ jxrlib/image/sys/strcodec.c
--- jxrlib/image/sys/strcodec.c.1~	2021-03-05 05:26:15.766718580 +0100
+++ jxrlib/image/sys/strcodec.c	2021-03-05 05:29:39.532376485 +0100
@@ -668,9 +668,7 @@ ERR detach_SB(SimpleBitIO* pSB)
 // WinCE ARM and Desktop x86
 #else
 // other platform
-#ifdef _BIG__ENDIAN_
-#define _byteswap_ulong(x)  (x)
-#else // _BIG__ENDIAN_
+#ifndef _BIG__ENDIAN_
 U32 _byteswap_ulong(U32 bits)
 {
     U32 r = (bits & 0xffu) << 24;
diff -up jxrlib/image/sys/strcodec.h.1~ jxrlib/image/sys/strcodec.h
--- jxrlib/image/sys/strcodec.h.1~	2021-03-05 05:26:15.766718580 +0100
+++ jxrlib/image/sys/strcodec.h	2021-03-05 05:29:24.680260899 +0100
@@ -64,7 +64,7 @@
 
 #ifndef UNREFERENCED_PARAMETER
 #define UNREFERENCED_PARAMETER(P) { (P) = (P); }
-#endif UNREFERENCED_PARAMETER
+#endif //UNREFERENCED_PARAMETER
 
 #ifdef UNDER_CE
 #define PLATFORM_WCE
@@ -676,3 +676,14 @@ void flushToByte(BitIOInfo* pIO);
 //    pIO->uiAccumulator = LOAD16(pIO->pbCurrent) & ((U32)(-1) >> pIO->cBitsUsed);\
 
 void OutputPerfTimerReport(CWMImageStrCodec *pState);
+
+#if (defined(WIN32) && !defined(UNDER_CE)) || (defined(UNDER_CE) && defined(_ARM_))
+// WinCE ARM and Desktop x86
+#else
+// other platform
+#ifdef _BIG__ENDIAN_
+#define _byteswap_ulong(x)  (x)
+#else // _BIG__ENDIAN_
+U32 _byteswap_ulong(U32 bits);
+#endif // _BIG__ENDIAN_
+#endif
diff -up jxrlib/jxrencoderdecoder/JxrDecApp.c.1~ jxrlib/jxrencoderdecoder/JxrDecApp.c
--- jxrlib/jxrencoderdecoder/JxrDecApp.c.1~	2021-03-05 05:26:15.766718580 +0100
+++ jxrlib/jxrencoderdecoder/JxrDecApp.c	2021-03-05 05:30:08.913603062 +0100
@@ -423,7 +423,7 @@ ERR WmpDecAppCreateEncoderFromExt(
     Call(GetTestEncodeIID(szExt, &pIID));
 
     // Create encoder
-    Call(PKTestFactory_CreateCodec(pIID, ppIE));
+    Call(PKTestFactory_CreateCodec(pIID, (void**)ppIE));
 
 Cleanup:
     return err;
diff -up jxrlib/jxrencoderdecoder/JxrEncApp.c.1~ jxrlib/jxrencoderdecoder/JxrEncApp.c
--- jxrlib/jxrencoderdecoder/JxrEncApp.c.1~	2021-03-05 05:26:15.767718588 +0100
+++ jxrlib/jxrencoderdecoder/JxrEncApp.c	2021-03-05 05:30:28.869755443 +0100
@@ -578,7 +578,7 @@ main(int argc, char* argv[])
 
     //================================
     Call(PKCreateCodecFactory(&pCodecFactory, WMP_SDK_VERSION));
-    Call(pCodecFactory->CreateCodec(&IID_PKImageWmpEncode, &pEncoder));
+    Call(pCodecFactory->CreateCodec(&IID_PKImageWmpEncode, (void**)&pEncoder));
 
     //----------------------------------------------------------------
     Call(PKCreateTestFactory(&pTestFactory, WMP_SDK_VERSION));
diff -up jxrlib/jxrgluelib/JXRGlueJxr.c.1~ jxrlib/jxrgluelib/JXRGlueJxr.c
--- jxrlib/jxrgluelib/JXRGlueJxr.c.1~	2021-03-05 05:26:15.767718588 +0100
+++ jxrlib/jxrgluelib/JXRGlueJxr.c	2021-03-05 05:30:50.957922741 +0100
@@ -28,7 +28,7 @@
 //*@@@---@@@@******************************************************************
 #include <limits.h>
 #include <JXRGlue.h>
-
+#include <wchar.h>
 
 static const char szHDPhotoFormat[] = "<dc:format>image/vnd.ms-photo</dc:format>";
 const U32 IFDEntryTypeSizes[] = { 0, 1, 1, 2, 4, 8, 1, 1, 2, 4, 8, 4, 8 };
diff -up jxrlib/jxrgluelib/JXRMeta.h.1~ jxrlib/jxrgluelib/JXRMeta.h
--- jxrlib/jxrgluelib/JXRMeta.h.1~	2021-03-05 05:26:15.768718597 +0100
+++ jxrlib/jxrgluelib/JXRMeta.h	2021-03-05 05:31:05.137029408 +0100
@@ -34,7 +34,7 @@
 
 #ifndef UNREFERENCED_PARAMETER
 #define UNREFERENCED_PARAMETER(P) { (P) = (P); }
-#endif UNREFERENCED_PARAMETER
+#endif //UNREFERENCED_PARAMETER
 
 //================================================================
 // Container
diff -up jxrlib/jxrtestlib/JXRTest.c.1~ jxrlib/jxrtestlib/JXRTest.c
--- jxrlib/jxrtestlib/JXRTest.c.1~	2021-03-05 05:26:15.768718597 +0100
+++ jxrlib/jxrtestlib/JXRTest.c	2021-03-05 05:32:50.540806067 +0100
@@ -198,7 +198,7 @@ ERR PKTestFactory_CreateDecoderFromFile(
     ERR err = WMP_errSuccess;
 
     char *pExt = NULL;
-    PKIID* pIID = NULL;
+    const PKIID* pIID = NULL;
 
     struct WMPStream* pStream = NULL;
     PKImageDecode* pDecoder = NULL;
@@ -214,7 +214,7 @@ ERR PKTestFactory_CreateDecoderFromFile(
     Call(CreateWS_File(&pStream, szFilename, "rb"));
 
     // Create decoder
-    Call(PKTestFactory_CreateCodec(pIID, ppDecoder));
+    Call(PKTestFactory_CreateCodec(pIID, (void**)ppDecoder));
     pDecoder = *ppDecoder;
 
     // attach stream to decoder
@@ -232,7 +232,7 @@ ERR PKCreateTestFactory(PKCodecFactory**
 
     UNREFERENCED_PARAMETER( uVersion );
 
-    Call(PKAlloc(ppCFactory, sizeof(**ppCFactory)));
+    Call(PKAlloc((void**)ppCFactory, sizeof(**ppCFactory)));
     pCFactory = *ppCFactory;
 
     pCFactory->CreateCodec = PKTestFactory_CreateCodec;
@@ -287,7 +287,7 @@ ERR PKTestDecode_Release(
 
     pID->fStreamOwner && pID->pStream->Close(&pID->pStream);
 
-    return PKFree(ppID);
+    return PKFree((void**)ppID);
 }
 
 ERR PKTestDecode_Create(
@@ -296,7 +296,7 @@ ERR PKTestDecode_Create(
     ERR err = WMP_errSuccess;
     PKTestDecode* pID = NULL;
 
-    Call(PKAlloc(ppID, sizeof(**ppID)));
+    Call(PKAlloc((void**)ppID, sizeof(**ppID)));
 
     pID = *ppID;
     pID->Initialize = PKTestDecode_Initialize;
diff -up jxrlib/jxrtestlib/JXRTestHdr.c.1~ jxrlib/jxrtestlib/JXRTestHdr.c
--- jxrlib/jxrtestlib/JXRTestHdr.c.1~	2021-03-05 05:26:15.768718597 +0100
+++ jxrlib/jxrtestlib/JXRTestHdr.c	2021-03-05 05:33:02.369891586 +0100
@@ -27,7 +27,7 @@
 //*@@@---@@@@******************************************************************
 #ifndef ANSI
 #define _CRT_SECURE_NO_WARNINGS
-#endif ANSI
+#endif //ANSI
 
 #include <stdlib.h>
 #include <string.h>
diff -up jxrlib/jxrtestlib/JXRTestPnm.c.1~ jxrlib/jxrtestlib/JXRTestPnm.c
--- jxrlib/jxrtestlib/JXRTestPnm.c.1~	2021-03-05 05:26:15.769718605 +0100
+++ jxrlib/jxrtestlib/JXRTestPnm.c	2021-03-05 05:33:15.873988842 +0100
@@ -27,7 +27,7 @@
 //*@@@---@@@@******************************************************************
 #ifndef ANSI
 #define _CRT_SECURE_NO_WARNINGS
-#endif ANSI
+#endif //ANSI
 
 #include <stdlib.h>
 
diff -up jxrlib/jxrtestlib/JXRTestTif.c.1~ jxrlib/jxrtestlib/JXRTestTif.c
--- jxrlib/jxrtestlib/JXRTestTif.c.1~	2021-03-05 05:26:15.769718605 +0100
+++ jxrlib/jxrtestlib/JXRTestTif.c	2021-03-05 05:33:36.363135670 +0100
@@ -909,8 +909,8 @@ ERR PKImageDecode_Release_TIF(PKTestDeco
 
     PKTestDecode *pID = *ppID;
 
-    Call(WMPFree(&pID->EXT.TIF.uStripOffsets));
-    Call(WMPFree(&pID->EXT.TIF.uStripByteCounts));
+    Call(WMPFree((void**)&pID->EXT.TIF.uStripOffsets));
+    Call(WMPFree((void**)&pID->EXT.TIF.uStripByteCounts));
 
     Call(PKTestDecode_Release(ppID));
 
